pkg/controllers/cloudmap_controller.go

```diff
@@ -252,37 +329,31 @@ func createDerivedServiceStruct(svc *model.Service, svcImport *v1alpha1.ServiceI
 		},
 		Spec: v1.ServiceSpec{
 			Type:  v1.ServiceTypeClusterIP,
-			Ports: extractServicePorts(svc),
+			Ports: extractServicePorts(endpoints),
 		},
 	}
 }
 
-func createEndpointSlicesStruct(svcImport *v1alpha1.ServiceImport, cloudMapSvc *model.Service, svc *v1.Service) []*discovery.EndpointSlice {
-	slices := make([]*discovery.EndpointSlice, 0)
-
+func createEndpointForSlice(svc *v1.Service, ip string) discovery.Endpoint {
 	t := true
 
-	endpoints := make([]discovery.Endpoint, 0)
-	for _, ep := range cloudMapSvc.Endpoints {
-		endpoints = append(endpoints, discovery.Endpoint{
-			Addresses: []string{ep.IP},
-			Conditions: discovery.EndpointConditions{
-				Ready: &t,
-			},
-			TargetRef: &v1.ObjectReference{
-				Kind:            "Service",
-				Namespace:       svc.Namespace,
-				Name:            svc.Name,
-				UID:             svc.ObjectMeta.UID,
-				ResourceVersion: svc.ObjectMeta.ResourceVersion,
-			},
-		})
+	return discovery.Endpoint{
+		Addresses: []string{ip},
+		Conditions: discovery.EndpointConditions{
+			Ready: &t,
+		},
+		TargetRef: &v1.ObjectReference{
+			Kind:            "Service",
+			Namespace:       svc.Namespace,
+			Name:            svc.Name,
+			UID:             svc.ObjectMeta.UID,
+			ResourceVersion: svc.ObjectMeta.ResourceVersion,
+		},
 	}
+}
 
-	// TODO split slices in case there are more than 1000 endpoints
-	// see https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/0752-endpointslices/README.md
-
-	slices = append(slices, &discovery.EndpointSlice{
+func createEndpointSliceStruct(svcImport *v1alpha1.ServiceImport, svc *v1.Service, endpoints []discovery.Endpoint, ports []discovery.EndpointPort) *discovery.EndpointSlice {
+	return &discovery.EndpointSlice{
```

Comment:
One TODO would be add the label `endpoints.kubernetes.io/over-capacity` when we reach 1000 endpoints.

Reply:
I believe since we're using slices, and keeping them small we should not have this problem.